Source/MQTTnet/Implementations/MqttTcpChannel.cs

```diff
@@ -274,6 +274,10 @@ bool InternalUserCertificateValidationCallback(object sender, X509Certificate x5
                 return certificateValidationHandler(eventArgs);
             }
 
+            if(_tcpOptions?.TlsOptions?.IgnoreCertificateChainErrors ?? false)
```

Comment:
Since the _SslPolicyErrors_ are flags I recommend to check for existence of "SslPolicyErrors.RemoteCertificateChainErrors" via "HasFlag()" and removing it when _IgnoreCertificateChainErrors _ is true. Then further properties can be added which will "remove" more flags from the validation.

Reply:
I used the AND operator along with the inversion of the bits to remove the flag. 
This will remove it if it exists and leave it unchanged if not.